common/types/observability_saved_object_attributes.ts

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObjectAttributes } from '../../../../src/core/types';
+import { SavedVisualization } from './explorer';
+
+export const VISUALIZATION_SAVED_OBJECT = 'observability-visualization';
+export const OBSERVABILTY_SAVED_OBJECTS = [VISUALIZATION_SAVED_OBJECT] as const;
+export const SAVED_OBJECT_VERSION = 1;
+
+export interface VisualizationSavedObjectAttributes extends SavedObjectAttributes {
+  title: string;
+  description: string;
+  version: number;
+  createdTimeMs: number;
+  savedVisualization: SavedVisualization;
+}
```

Comment:
I see that we are using the public service for SavedObjects. The server service has more options for something like version: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/416eb508b3b22fe81231b726dcfce09e3aad9bfb/src/core/server/saved_objects/service/saved_objects_client.ts#L47-L71
Did we take a look at it?

Reply:
version can be passed as a create option in server but not in frontend. https://github.com/opensearch-project/opensearch-dashboards/blob/11b98ec05483269917c335fcb1900bf98da8cac6/src/core/public/saved_objects/saved_objects_client.ts#L54

If we want to use it then saved object client should be called in server? But this seems to be the version of a specific object that will change on each update, not sure if it would be useful for us

```
  /**
   * An opaque version number which changes on each successful write operation.
   * Can be used in conjunction with `overwrite` for implementing optimistic concurrency control.
   **/
  version?: string;
```

Reply:
That isn't required then. 👍🏽